### ISSUE TITLE
segmentation fault when using some channels enum over EBUR128_DUAL_MONO

### ISSUE THREAD

**Issue by audionuma** [jiixyj/libebur128#77]
Hello,
here's a small example (built against current master)

```C
#include <stdlib.h>
#include "ebur128.h"

int main(int ac, const char *av[])
{
    ebur128_state* st = malloc(sizeof(ebur128_state*));
    st = ebur128_init(1, 48000, EBUR128_MODE_I);
    ebur128_set_channel(st, 0, EBUR128_Mp060);
    int samples[48000] = {0};

    // this call will raise a segmentation fault
    ebur128_add_frames_int(st, samples, 48000);
    
    ebur128_destroy(&st);
    free(st);
}
```
It seems it is related to the way the variable `ci`is used in the `EBUR128_FILTER` code.
See line 585 in ebur128.c : if `channel_map[c]` is greater than 5, `ci` will be greater than 4, and is then used as an index to a size 5 array.

**Comment by audionuma** [comment]
Hum ... a very dirty workaround (ignoring potential bad side effects) :
ebur128.c, line 61
```C
  /** BS.1770 filter state. */
  double v[EBUR128_Bm045 - 1][5];
```
and
ebur128.c, line 248
```C
  for (i = 0; i < EBUR128_Bm045 - 1; ++i) {
    for (j = 0; j < 5; ++j) {
      st->d->v[i][j] = 0.0;
    }
  }
```

**Comment by jiixyj** [comment]
I wonder why we index the filter state by channel enum instead of just the channel index itself... I'll have a look at it.

**Comment by audionuma** [comment]
See https://github.com/audionuma/libebur128/commit/8622f1f67def1fa1119b45dbe9a9205e69f9dc2c for an attempt to solve the issue.

cons : allocate filter_state for unused channels, not sure of my handling of the ERROR_CHECK macro.

**Comment by jiixyj** [comment]
Looks good so far! Could you also modify `ebur128_change_parameters` so that the filter states are correctly reallocated if the number of channels changes?

**Comment by jiixyj** [comment]
I think we should limit the maximum number of channels to something relatively sane (like 1024) so that the multiplication inside the `malloc` call cannot overflow.

**Comment by jiixyj** [comment]
I've merged your work into the `filter-channel-merge` branch. I fixed up the cleanup/free code a bit, but otherwise it looks good.

**Comment by jiixyj** [comment]
I've merged it into `master` now.

**Comment by jiixyj** [comment]
Finally released with 1.2.5! Thanks again for the fix.
